SU2_CFD/src/solver_structure.cpp

```diff
@@ -4825,6 +4825,365 @@ void CSolver::Read_InletFile_ASCII(CGeometry *geometry, CConfig *config, string
   
 }
 
+void CSolver::PrintNewInletData (passivedouble *Inlet_InterpolatedData, unsigned long nVertex,string Marker_Tag)
+{
+
+ofstream myfile;
+myfile.precision(16);
+myfile.open("Interpolated_Data_"+Marker_Tag+".dat",ios_base::out);
+
+
+if (myfile.is_open())
+{  
+            for (unsigned long iVertex = 0; iVertex < nVertex; iVertex++) {
+
+                  for  (unsigned long iVar=0; iVar < (maxCol_InletFile+nDim); iVar++)
+                    {
+                      myfile<<Inlet_InterpolatedData[iVertex*(maxCol_InletFile+nDim)+iVar]<<"\t";
+                    }
+                  myfile<<endl;
+                }
+        myfile.close();
+}
+  else
+    cout<<"file cannot be opened"<<endl;
+}
+
+
+void CSolver::LoadSpanwiseInletProfile(CGeometry **geometry,
+                                CSolver ***solver,
+                                CConfig *config,
+                                int val_iter,
+                                unsigned short val_kind_solver,
+                                unsigned short val_kind_marker)
```

Comment:
This method is quite long, it is well documented but please consider breaking it up in more manageable functions.
Especially the part that interpolates to coarse levels sounds like something that can be reused elsewhere.

Reply:
The existing subroutine was "LoadInletProfile" and I reused the same part of the code here to get down to coarser grids.

Reply:
Ok can you please check if there is an easy way to factor that multigrid part out of both "LoadInlet" methods, so that both call a function to do that job instead of having the code copied.

Reply:
I wanted to look into that but also didn't want to change the existing code. I will see if that can be done.